File: solutions/4353-largest-integer-with-given-digit-sum/largest-integer-with-given-digit-sum.js
function largestInteger(n, s) {
    let res = 0;
    for (const min = Math.min; s; s -= min(s, 9)) {
        if (!n--) return -1;
        res = res * 10 + min(s, 9);
    }
    return res * 10 ** n;
}

File: solutions/4353-largest-integer-with-given-digit-sum/largest-integer-with-given-digit-sum.py
class Solution:
    def largestInteger(self, n: int, s: int) -> int:
        res = 0
        while s > 0:
            if n == 0:
                return -1
            n -= 1
            d = min(s, 9)
            res = res * 10 + d
            s -= d
        return res * 10**n